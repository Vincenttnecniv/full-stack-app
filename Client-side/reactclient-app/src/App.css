body{
    background-image: linear-gradient(to right, #0f0c29, #302b63, #24243e);
}

h1{
    color: #48d1cc !important;
     
    text-align: center !important;
}

form{
text-align: center !important;
}

h2,h3{
    color: 	#48d1cc ;
    text-align: center !important;
}

td{
    width:400px; 
    height:100px;
    color: #fff !important;
}

th{
    
    color: #fff !important;
}

.trackID{
    display: none
}

p{
    font-size: 20px;
    color: #fff !important;
    text-align: center !important;
}