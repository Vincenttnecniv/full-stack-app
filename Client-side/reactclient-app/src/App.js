import './App.css';
import HandleApiData from "./Components/HandleApiData"; 


function App() {
  return (
    <div className="App">
      <header className="App-header">
        <HandleApiData />
      </header>
    </div>
  );
}

export default App;