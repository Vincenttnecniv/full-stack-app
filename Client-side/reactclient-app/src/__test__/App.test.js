import renderer from "react-test-renderer";
import { render, screen, cleanup } from "@testing-library/react";
import App from "../App";
//Heres where I run my tests for app.
describe("App component", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

test("Should return a 404 when invalid route", () => {
  render(<App />);
  const appElement = screen.getByTestId("app-1");
  expect(appElement).toBeInTheDocument();
});
