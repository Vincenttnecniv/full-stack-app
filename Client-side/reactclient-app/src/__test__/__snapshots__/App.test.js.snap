// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches the snapshot 1`] = `
<div
  className="App"
  data-testid="app-1"
>
  <h1>
    For your media to render please give your qeury and media type below !
  </h1>
  <br />
  <div>
    <br />
    <div>
      <form
        onSubmit={[Function]}
      >
        <input
          onChange={[Function]}
          placeholder="Enter Query"
          type="text"
          value="dax"
        />
        <br />
        <input
          onChange={[Function]}
          placeholder="Enter Media"
          required={true}
          type="text"
          value="music"
        />
        <br />
        <input
          type="submit"
          value="Submit"
        />
        <br />
        <br />
        <br />
        <br />
        <br />
        <h2>
          The results of your media is displayed below, select your favourites !
        </h2>
        <br />
        <p>
          All data will display here
        </p>
      </form>
      <br />
      <br />
      <br />
      <h3>
        Enter the index of your favorite track (starting from 0)
      </h3>
      <form
        onSubmit={[Function]}
      >
        <input
          onChange={[Function]}
          placeholder="Enter fav index"
          required={true}
          type="number"
          value=""
        />
        <br />
        <br />
        <input
          type="submit"
          value="Add"
        />
      </form>
      <br />
      <br />
      <h3>
        Press delete to delete all tracks
      </h3>
      <form
        onSubmit={[Function]}
      >
        <input
          type="submit"
          value="Delete"
        />
      </form>
      <br />
      <h1>
        Your favourites
      </h1>
      <br />
      <br />
      <p>
        All favourites will display here
      </p>
    </div>
  </div>
</div>
`;
